### PR TITLE
Allow /log?pod=foo&id=123 from deck.

### PR DESCRIPTION
**Pull request by spxtr**
Part 1 of #3402. I'm not super happy about this code, and will want to do a refactor soon.

### CODE REVIEW

File: prow/cmd/deck/jobs.go [hunk 1]
```diff
@@ -35,37 +35,39 @@ const (
...
-	Started     string `json:"started"`
-	Finished    string `json:"finished"`
-	Duration    string `json:"duration"`
-	State       string `json:"state"`
-	Description string `json:"description"`
-	URL         string `json:"url"`
-	PodName     string `json:"pod_name"`
-	Agent       string `json:"agent"`
-	ProwJob     string `json:"prow_job"`
+	Type        string            `json:"type"`
```
Comment (rmmh) [inline prow/cmd/deck/jobs.go]: cool diff gofmt

File: prow/cmd/deck/jobs.go [hunk 2]
```diff
@@ -35,37 +35,39 @@ const (
...
-	jc      *jenkins.Client
-	jobs    []Job
-	jobsMap map[string]Job // pod name -> Job
-	mut     sync.Mutex
+	kc        *kube.Client
+	pkc       *kube.Client
+	jc        *jenkins.Client
+	jobs      []Job
+	jobsMap   map[string]Job                     // pod name -> Job
+	jobsIDMap map[string]map[string]kube.ProwJob // job name -> id -> ProwJob
```
Comment (rmmh) [inline prow/cmd/deck/jobs.go]: any reason not to map name + "\n" + id to ProwJob?

File: prow/cmd/deck/jobs.go (hunk 1 above)
Comment (spxtr) [inline prow/cmd/deck/jobs.go]: yeah... it does that.

File: prow/cmd/deck/jobs.go (hunk 2 above)
Comment (spxtr) [inline prow/cmd/deck/jobs.go]: That would be okay but I'm not a huge fan.

**Pull request merged**